mvn-cli.js

```diff
@@ -1,42 +1,63 @@
 const { docker } = require("@kaholo/plugin-library");
+const { join: joinPaths } = require("path");
+const { homedir: getHomeDirectory } = require("os");
 const {
   exec,
   assertPathExistence,
 } = require("./helpers");
 const {
   MAVEN_DOCKER_IMAGE,
   MAVEN_CLI_NAME,
+  MAVEN_CACHE_DIRECTORY_NAME,
 } = require("./consts.json");
 
 async function execute({ command, workingDirectory }) {
-  const volumeConfigsMap = new Map();
+  const dockerCommandBuildOptions = {
+    command: docker.sanitizeCommand(command, MAVEN_CLI_NAME),
+    image: MAVEN_DOCKER_IMAGE,
+  };
+
+  const mavenAgentCachePath = joinPaths(getHomeDirectory(), MAVEN_CACHE_DIRECTORY_NAME);
+  const mavenCacheVolumeDefinition = docker.createVolumeDefinition(mavenAgentCachePath);
+  // Change mount point to maven cache path
+  mavenCacheVolumeDefinition.mountPoint.value = joinPaths("/root", MAVEN_CACHE_DIRECTORY_NAME);
+
+  let dockerEnvironmentalVariables = {
+    [mavenCacheVolumeDefinition.mountPoint.name]: mavenCacheVolumeDefinition.mountPoint.value,
+  };
+  let shellEnvironmentalVariables = {
+    ...dockerEnvironmentalVariables,
+    [mavenCacheVolumeDefinition.path.name]: mavenCacheVolumeDefinition.path.value,
+  };
+
+  const volumeDefinitionsArray = [mavenCacheVolumeDefinition];
+
   if (workingDirectory) {
     await assertPathExistence(workingDirectory);
-    volumeConfigsMap.set("workingDirectory", docker.createVolumeConfig(workingDirectory));
-  }
+    const volumeDefinition = docker.createVolumeDefinition(workingDirectory);
 
-  const volumeConfigsArray = [...volumeConfigsMap.values()];
-  const {
-    environmentVariablesRequiredByDocker,
-    environmentVariablesRequiredByShell,
-  } = docker.extractEnvironmentVariablesFromVolumeConfigs(volumeConfigsArray);
+    dockerEnvironmentalVariables = {
```

Comment:
New parameter can be assigned directly to an existing object. No need to recreate it just to achieve that.